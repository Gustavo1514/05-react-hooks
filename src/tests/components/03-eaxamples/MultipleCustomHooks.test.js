const { shallow } = require('enzyme')
const {
  default: MultipleCustomHooks
} = require('../../../components/03-examples/MultipleCustomHooks')
const { useFetch } = require('../../../hooks/useFetch')
const { useCounter } = require('../../../hooks/useCounter')
jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('Pruebas sobre <MultipleCustomHooks/>', () => {
  //   beforeEach(() => {

  //   })
  test('debe mostrarse correctamente ', () => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {}
    })
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    })
    const wrapper = shallow(<MultipleCustomHooks />)
    expect(wrapper).toMatchSnapshot()
  })

  test('debe mostrar la informacion', () => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {}
    })
    useFetch.mockReturnValue({
      data: [
        {
          author: 'Gustav',
          quote: 'Hola Mundo'
        }
      ],
      loading: false,
      error: null
    })
    const wrapper = shallow(<MultipleCustomHooks />)
    expect(wrapper.find('.alert').exists()).toBe(false)
    expect(
      wrapper
        .find('.mb-0')
        .text()
        .trim()
    ).toBe('Hola Mundo')
    expect(
      wrapper
        .find('footer')
        .text()
        .trim()
    ).toBe('Gustav')

    // console.log(wrapper.html())
  })
})
