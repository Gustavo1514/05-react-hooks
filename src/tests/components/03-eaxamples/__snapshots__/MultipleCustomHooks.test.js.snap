// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <MultipleCustomHooks/> debe mostrarse correctamente  1`] = `
<div>
  <h1>
    Breaing Bad Cuotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Nueva Frase
  </button>
</div>
`;
